### PR TITLE
[wizard] Make upgrading to Xbase seamless

### PR DESCRIPTION
**Pull request by oehme**

### CODE REVIEW

File: plugins/org.eclipse.xtext.xbase.ide/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -8,7 +8,7 @@ Bundle-RequiredExecutionEnvironment: JavaSE-1.6
 Bundle-ActivationPolicy: lazy
 Bundle-Localization: plugin
 Require-Bundle: org.eclipse.xtext.ide,
- org.eclipse.xtext.xbase,
+ org.eclipse.xtext.xbase;visibility:=reexport,
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xbase.ide/META-INF/MANIFEST.MF]: I'm not convinced by this re-export. Why is that necessary?

Comment (oehme) [inline plugins/org.eclipse.xtext.xbase.ide/META-INF/MANIFEST.MF]: I understand that concern with bundles outside our control, but why would anybody use xbase.ide without using xbase? I thought that was the same rationale why xbase.ui exports xbase too.

If we don't want this, then we'll have to add more require bundles to the XbaseGeneratorFragment2

Comment (szarnekow) [inline plugins/org.eclipse.xtext.xbase.ide/META-INF/MANIFEST.MF]: Agreed. I was somehow concerned about re-exported re-exports, but those will be pulled into the mydsl.ide bundle via an explicit xbase dependency, too.

File: plugins/org.eclipse.xtext.xtext.wizard/xtend-gen/org/eclipse/xtext/xtext/wizard/WebProjectDescriptor.java [hunk 2]
```diff
@@ -73,8 +73,10 @@ public boolean isPartOfMavenBuild() {
     final LinkedHashSet<ExternalDependency> deps = CollectionLiterals.<ExternalDependency>newLinkedHashSet();
     Set<ExternalDependency> _externalDependencies = super.getExternalDependencies();
     Iterables.<ExternalDependency>addAll(deps, _externalDependencies);
-    ExternalDependency _createXtextDependency = ExternalDependency.createXtextDependency("org.eclipse.xtext.web.servlet");
+    ExternalDependency _createXtextDependency = ExternalDependency.createXtextDependency("org.eclipse.xtext.xbase.web");
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.wizard/xtend-gen/org/eclipse/xtext/xtext/wizard/WebProjectDescriptor.java]: Should ExternalDependency have a meaningful equals/hashcode?

Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.wizard/xtend-gen/org/eclipse/xtext/xtext/wizard/WebProjectDescriptor.java]: That would be helpful for testing. It is a mutable class though, that's why I left equals/hashCode out

Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.wizard/xtend-gen/org/eclipse/xtext/xtext/wizard/WebProjectDescriptor.java]: Since we put them into a set to avoid duplicates, they have to have a proper equals semantics, don't they?
